### PR TITLE
Make it possible to configure adversarial training

### PR DESCRIPTION
**Pull request by goodfeli**

### CODE REVIEW

File: cleverhans/loss.py [hunk 1]
```diff
@@ -84,20 +109,38 @@ class CrossEntropy(Loss):
   :param model: Model instance, the model on which to apply the loss.
   :param smoothing: float, amount of label smoothing for cross-entropy.
   :param attack: function, given an input x, return an attacked x'.
+  :param pass_y: bool, if True pass y to the attack
   """
-  def __init__(self, model, smoothing=0., attack=None, **kwargs):
+  def __init__(self, model, smoothing=0., attack=None, pass_y=False,
+               adv_coeff=0.5, attack_params=None,
```
Comment (catherio) [inline cleverhans/loss.py]: can you add a docstring for adv_coeff and attack_params?

File: cleverhans/loss.py [hunk 2]
```diff
@@ -24,7 +26,30 @@ def __init__(self, model, hparams=None, attack=None):
     :param attack: callable, the attack function for adv. training.
```
Comment (catherio) [inline cleverhans/loss.py]: shouldn't this NOT be "callable" in the docstring anymore?

File: cleverhans/loss.py [hunk 3]
```diff
@@ -84,20 +109,38 @@ class CrossEntropy(Loss):
...
     if self.attack is not None:
-      x = x, self.attack(x)
+      attack_params = copy.copy(self.attack_params)
+      if attack_params is None:
+        attack_params = {}
+      if self.pass_y:
+        attack_params['y'] = y
+      x = x, self.attack.generate(x, **attack_params)
+      coeffs = [1. - self.adv_coeff, self.adv_coeff]
+      assert len(coeffs) == len(x)
```
Comment (catherio) [inline cleverhans/loss.py]: why is this assert useful? it seems definitionally true from the previous two lines

File: cleverhans/loss.py [hunk 4]
```diff
@@ -84,20 +109,38 @@ class CrossEntropy(Loss):
...
+      if self.adv_coeff == 1.:
+        x = (x[1],)
+        coeffs = (coeffs[1],)
     else:
       x = tuple([x])
+      coeffs = [1.]
+      assert len(coeffs) == len(x)
```
Comment (catherio) [inline cleverhans/loss.py]: same - seems impossible to violate? is this just to help future implementers?

Comment (goodfeli) [reply cleverhans/loss.py]: Thanks, removed it

File: cleverhans/loss.py (hunk 3 above)
Comment (goodfeli) [reply cleverhans/loss.py]: Thanks, removed it

File: cleverhans/loss.py (hunk 2 above)
Comment (goodfeli) [reply cleverhans/loss.py]: Thanks, updated it to specify an Attack

File: cleverhans/loss.py (hunk 1 above)
Comment (goodfeli) [reply cleverhans/loss.py]: thanks, added docstrings

**Comment by goodfeli** [comment]
Ready for re-review

**Pull request merged**